File: 1.0/src/cz/mg/vulkan/PFNvkGetPipelineCacheData.java
package cz.mg.vulkan;

public class PFNvkGetPipelineCacheData extends VkFunctionPointer {
    public PFNvkGetPipelineCacheData() {
    }

    protected PFNvkGetPipelineCacheData(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkGetPipelineCacheData(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkGetPipelineCacheData(long value) {
        setValue(value);
    }

    public PFNvkGetPipelineCacheData(VkInstance instance) {
        super(instance, new VkString("vkGetPipelineCacheData"));
    }

    public int call(VkDevice device, VkPipelineCache pipelineCache, VkSize pDataSize, VkObject pData){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pipelineCache != null ? pipelineCache.getVkAddress() : VkPointer.getNullAddressNative(), pDataSize != null ? pDataSize.getVkAddress() : VkPointer.NULL, pData != null ? pData.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, long device, long pipelineCache, long pDataSize, long pData);

}

File: 1.0/src/c/cz/mg/vulkan/PFN_vkGetPipelineCacheData.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkGetPipelineCacheData_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pipelineCache, jlong pDataSize, jlong pData){
    (void)env;
    (void)jc;
    PFN_vkGetPipelineCacheData f = (PFN_vkGetPipelineCacheData)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkPipelineCache*)jniLongToPointer(pipelineCache)),
        ((size_t*)jniLongToPointer(pDataSize)),
        ((void*)jniLongToPointer(pData))
    );
}

